cmd/promtool/main.go

```diff
@@ -263,24 +267,25 @@ func checkConfig(filename string) ([]string, error) {
 			return nil, err
 		}
 
-		for _, kd := range scfg.ServiceDiscoveryConfig.KubernetesSDConfigs {
-			if err := checkTLSConfig(kd.HTTPClientConfig.TLSConfig); err != nil {
-				return nil, err
-			}
-		}
-
-		for _, filesd := range scfg.ServiceDiscoveryConfig.FileSDConfigs {
-			for _, file := range filesd.Files {
-				files, err := filepath.Glob(file)
-				if err != nil {
+		for _, c := range scfg.ServiceDiscoveryConfigs {
+			switch c := c.(type) {
+			case *kubernetes.SDConfig:
```

Comment:
Yeah, I wasn't happy with this either, but again I didn't want to change the existing behavior with this PR.

I would like to hold this off for a followup, if that's alright. I think we could add a `Validate() error` method to the `discovery.Config` interface or it could be on its own and we could check here if the `discovery.Config` implements `interface { Validate() error }`?

Reply:
I went ahead and added `Validate() error` to the `discovery.Config` interface.

Reply:
Yes, this needs to be solved generally.